src/MessageBroker.php

```diff
@@ -147,16 +147,19 @@ public function __destruct() {
    *
    * @param string $payload
    *  Data to wrap in the message.
-   *
    * @param int $deliveryMode
    *  1: non-persistent, faster but no logging to disk, ~ 3x
    *  2: persistent, write a copy of the message to disk
    *
    * The related queue must also be set to durable for the setting
    * to work. If the server crashes, persistent messages will be recovered.
-   * Crash tolerance at a price.
+   * Crash tolerance comes at a price.
+   * @param string $routingKey
+   *  The key or pattern the message will use for submission to the exchange for distribution to the
+   *  attached queues. Depending on the exchange type the key defines what queues get a copy of
+   *  the message.
    */
-  public function publishMessage($payload, $deliveryMode = 1) {
+  public function publishMessage($payload, $deliveryMode = 1, $routingKey) {
```

Comment:
It's a bit of a bummer to have the optional parameter stuck in the middle here. Do you think it would make sense to make a new `publish($payload, $routingKey, $deliveryMode = 1)` method and deprecate this one for a simpler API?